programs/perpetuals/src/state/custody.rs

```diff
@@ -289,6 +358,121 @@ impl Custody {
 
         Ok(())
     }
+
+    pub fn get_collective_position(&self, side: Side) -> Result<Position> {
+        let stats = if side == Side::Long {
+            &self.long_positions
+        } else {
+            &self.short_positions
+        };
+        if stats.open_positions > 0 {
+            Ok(Position {
+                side,
+                price: math::checked_as_u64(math::checked_div(
+                    stats.weighted_leverage,
+                    stats.total_leverage,
+                )?)?,
+                size_usd: stats.size_usd,
+                collateral_usd: stats.collateral_usd,
+                unrealized_loss_usd: stats.cumulative_interest,
+                cumulative_interest_snapshot: stats.cumulative_interest_snapshot,
+                locked_amount: stats.locked_amount,
+                ..Position::default()
+            })
+        } else {
+            Ok(Position::default())
+        }
+    }
+
+    pub fn add_position(&mut self, position: &Position, curtime: i64) -> Result<()> {
+        // compute accumulated interest
+        let collective_position = self.get_collective_position(position.side)?;
+        let interest = self.get_interest_amount_usd(&collective_position, curtime)?;
+
+        // update positions
+        let stats = if position.side == Side::Long {
+            &mut self.long_positions
+        } else {
+            &mut self.short_positions
+        };
+
+        stats.cumulative_interest = math::checked_add(stats.cumulative_interest, interest)?;
+        stats.cumulative_interest_snapshot = position.cumulative_interest_snapshot;
+
+        stats.open_positions = math::checked_add(stats.open_positions, 1)?;
+        stats.collateral_usd = math::checked_add(stats.collateral_usd, position.collateral_usd)?;
+        stats.size_usd = math::checked_add(stats.size_usd, position.size_usd)?;
+        stats.locked_amount = math::checked_add(stats.locked_amount, position.locked_amount)?;
+
+        let leverage = position.get_initial_leverage()? as u128;
+        stats.weighted_leverage = math::checked_add(
+            stats.weighted_leverage,
+            math::checked_mul(position.price as u128, leverage)?,
+        )?;
+        stats.total_leverage = math::checked_add(stats.total_leverage, leverage)?;
+
+        Ok(())
+    }
+
+    pub fn remove_position(&mut self, position: &Position, curtime: i64) -> Result<()> {
+        // compute accumulated interest
+        let collective_position = self.get_collective_position(position.side)?;
+        let interest = self.get_interest_amount_usd(&collective_position, curtime)?;
+        let cumulative_interest_snapshot = self.get_cumulative_interest(curtime)?;
+
+        // update stats
+        let stats = if position.side == Side::Long {
+            &mut self.long_positions
+        } else {
+            &mut self.short_positions
+        };
+
+        if stats.open_positions == 1 {
+            *stats = PositionStats::default();
```

Comment:
why? if the last position is closed then we reset cumulative interest as well, because it is not outstanding anymore but realized and will be accounted for in AUM calc